launchpod/angular-launchpod/backend/src/main/java/com/google/launchpod/servlets/FileUploadServlet.java

```diff
@@ -0,0 +1,291 @@
+package com.google.launchpod.servlets;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.MultipartConfig;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PropertyContainer;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.auth.appengine.AppEngineCredentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.launchpod.data.RSS;
+import com.google.launchpod.data.UserFeed;
+import com.google.launchpod.data.MP3;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.PostPolicyV4;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.google.common.base.Strings;
+
+@WebServlet("/create-by-upload")
+@MultipartConfig
+public class FileUploadServlet extends HttpServlet {
```

Comment:
we should add tests for FileUploadServlet. feel free to do in another PR

Reply:
There are tests, I may have excluded them from the most recent commit. I'll add them back in now